src/ray/core_worker/task_event_buffer.cc

```diff
@@ -137,17 +137,22 @@ Status TaskEventBufferImpl::Start(bool auto_flush) {
     sigaddset(&mask, SIGTERM);
     pthread_sigmask(SIG_BLOCK, &mask, NULL);
 
+#endif
+#ifdef __linux__
     // Decrease the thread priority to allow worker threads to run.
+    // Skipping it for non-linux since niceness is applied to all threads on a process.
     int new_nice = std::min(
         RayConfig::instance().worker_niceness() + kTaskEventBufferAdditionalNice, 19);
     new_nice = nice(new_nice);
-    if (new_nice == -1) {
+    auto old_errno = errno;
```

Comment:
I believe this should be `nice(kTaskEventBufferAdditionalNice)` instead of `nice(worker_niceness + kTaskEventBufferAdditionalNice)`, as the thread will have the default worker niceness when it is created (and `nice` increments the niceness).

But I think this shouldn't break anything -- we may just end up with an overly nice thread. In the case where the worker thread has a high niceness, then we will see a performance regression.

I think we should file a P1 for this and merge this PR!

Reply:
oh yeah, I am feeling stupid about this. We should probably get it first and then add it... 

But yeah, since the cap is 19 for a thread, this essentially means the flushing thread is always the lowest.

Reply:
Actually - the worker niceness is set after this code path: 
https://github.com/ray-project/ray/blob/69a14e764891d62a1df9e1ef14ba7a736308b0e2/src/ray/core_worker/core_worker.cc#L599-L606

And this thread set up is done before that here: https://github.com/ray-project/ray/blob/69a14e764891d62a1df9e1ef14ba7a736308b0e2/src/ray/core_worker/core_worker.cc#L225-L232

So I think we do need to increment with the worker_niceness included.

Reply:
But I think the get first and then increment logic should still be the right way of doing it.

Reply:
Great!